Build fails: setup.py cannot find requirements.txt during PyPI publish

The GitHub Actions workflow for publishing to PyPI fails because setup.py tries to read requirements.txt at build time. During the isolated build process (PEP 517/518), requirements.txt may not be present or accessible, resulting in a `FileNotFoundError`.

**Error excerpt:**
```
FileNotFoundError: [Errno 2] No such file or directory: 'requirements.txt'
ERROR Backend subprocess exited when trying to invoke get_requires_for_build_wheel
Error: Process completed with exit code 1.
```

### Why this happens

- The build backend creates an isolated environment and may not include all project files.
- Reading requirements.txt in setup.py is not reliable for packaging and publishing.

### Solution

- Move all runtime dependencies directly into the `install_requires` list in setup.py.
- Only include actual runtime dependencies (not dev/test tools like `pytest`).
- Keep requirements.txt for development and testing, but do not rely on it in setup.py.

**Example:**
```python
setup(
    # ...other args...
    install_requires=[
        "google-api-python-client",
        "python-dotenv",
        "mwclient",
        "pycountry",
    ],
    # ...other args...
)
```

**References:**
- [Python Packaging User Guide: Configuring Metadata](https://packaging.python.org/en/latest/tutorials/packaging-projects/#configuring-metadata)
- [PEP 517/518: Build System Interface](https://www.python.org/dev/peps/pep-0517/)

---

**Action needed:**  
Update setup.py to list dependencies directly in `install_requires` and remove any code that reads requirements.txt at build time. This will make the package build reliably in CI and for PyPI.

--- a/.github/workflows/python-publish.yml
+++ b/.github/workflows/python-publish.yml
@@ -27,12 +27,20 @@ jobs:
         with:
           python-version: "3.x"
 
+      - name: Install Poetry
+        run: |
+          curl -sSL https://install.python-poetry.org | python3 -
+          export PATH="$HOME/.local/bin:$PATH"
+
+      - name: Install dependencies
+        run: |
+          export PATH="$HOME/.local/bin:$PATH"
+          poetry install --no-interaction --no-root
+
       - name: Build release distributions
         run: |
-          # NOTE: put your own distribution build steps here.
-          python -m pip install build
-          python -m pip install -r requirements.txt
-          python -m build
+          export PATH="$HOME/.local/bin:$PATH"
+          poetry build
 
       - name: Upload distributions
         uses: actions/upload-artifact@v4

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,7 +1,25 @@
-[build-system]
-requires = [
-    "setuptools>=40.8.0",
-    "wheel",
-    "python-dotenv"
+[project]
+name = "youtubewikibot"
+version = "0.1.1"
+description = "A bot to create a database of the most popular creators on YouTube."
+authors = [
+    {name = "Luis Imperial",email = "imperialluisanton@outlook.ph"}
 ]
-build-backend = "setuptools.build_meta"
+license = {text = "MIT"}
+readme = "README.md"
+requires-python = ">=3.10"
+dependencies = [
+    "google-api-python-client (>=2.173.0,<3.0.0)",
+    "python-dotenv (>=1.1.0,<2.0.0)",
+    "mwclient (>=0.11.0,<0.12.0)",
+    "pycountry (>=24.6.1,<25.0.0)"
+]
+
+
+[build-system]
+requires = ["poetry-core>=2.0.0,<3.0.0"]
+build-backend = "poetry.core.masonry.api"
+
+[tool.poetry.group.dev.dependencies]
+pytest = "^8.4.1"
+

--- a/setup.py
+++ b/setup.py
@@ -14,7 +14,7 @@ AUTHOR_EMAIL = os.getenv('AUTHOR_EMAIL', "default@example.com")
 
 setup(
     name='YouTubeWikiBot',
-    version='0.1',
+    version='0.1.1',
     author='Luis Imperial',
     author_email=AUTHOR_EMAIL,
     description='A bot to create a database of the most popular creators on YouTube.',

